apps/storefront/src/pages/Dashboard/index.tsx

```diff
@@ -128,6 +141,18 @@ function Dashboard(props: PageProps) {
     });
   };
 
+  const onStartMasquerade = (companyId: number) => {
+    try {
+      const cartEntityId = z.string().parse(Cookies.get('cartId'));
+      setCartEntityId(cartEntityId);
```

Comment:
After some thought, we might want to read the cartId directly in the startActing method, this way we don't have to keep track of two variables.

```tsx
const cartEntityId = Cookies.get('cartId');

if (cartEntityId) {
  await deleteCart({ deleteCartInput: { cartEntityId } });
  Cookies.remove('cartId');
  store.dispatch(setCartNumber(0));
  // setCartEntityId(''); this should probably go in the `handleConfirm`, as it is only specific to the modal being open.
}
```

This separates the action of acting from the action of confirming.

`startActing` will always remove the existing cart, we only change whether we want confirmation or not.